scripts/zones/Arrapago_Reef/npcs/_1ie.lua

```diff
@@ -0,0 +1,64 @@
+-----------------------------------
+-- Area: Arrapago Reef
+-- Door: Iron Gate (Lamian Fang Key)
+-- !pos 580 -17 120
+-----------------------------------
+package.loaded["scripts/zones/Arrapago_Reef/TextIDs"] = nil
+-----------------------------------
+require("scripts/zones/Arrapago_Reef/TextIDs")
+require("scripts/globals/npc_util")
+require("scripts/globals/status")
+-----------------------------------
+
+function onTrade(player,npc,trade)
+    if npc:getAnimation() == dsp.anim.CLOSE_DOOR then
+        if npcUtil.tradeHas(trade, 2219) then
+            npc:openDoor(dsp.anim.ANIMATION_OPEN_DOOR)
+            player:messageSpecial(KEY_BREAKS,2219)
+            player:confirmTrade()
+        elseif npcUtil.tradeHas(trade, 1022) and player:getMainJob() == dsp.job.THF then -- thief's tools
+            if math.random(1,2) == 1 then -- TODO: figure out actual percentage chance to pick locks; 50% for now
+                player:messageSpecial(LOCK_SUCCESS,1022)
+                npc:openDoor(dsp.anim.ANIMATION_OPEN_DOOR)
```

Comment:
dsp.anim.ANIMATION_OPEN_DOOR would be dsp.anim.OPEN_DOOR, but also npc:openDoor() seems to take as its only argument time in seconds.  (if omitted, defaults to 7 seconds)

Reply:
yea, i did use the wrong argument... that's the argument for setAnimation... :man_facepalming: